app/src/main/java/io/apicurio/registry/mt/TenantMetadataService.java

```diff
@@ -40,19 +63,116 @@
+        }
+
+        if (properties.getTenantManagerUrl().isEmpty()) {
+            throw new DeploymentException("Unsupported configuration, \"registry.enable.multitenancy\" is enabled " +
+                    "but the no \"registry.tenant.manager.url\" is provided");
+        }
+
+        Map<String, Object> clientConfigs = new HashMap<>();
+        if (properties.getTenantManagerCAFilePath().isPresent() && !properties.getTenantManagerCAFilePath().get().isBlank()) {
+            clientConfigs.put(ApicurioClientConfig.APICURIO_REQUEST_CA_BUNDLE_LOCATION, properties.getTenantManagerCAFilePath().get());
+        }
+
+        Auth auth = null;
+        if (properties.isTenantManagerAuthEnabled()) {
+
+            if (properties.getTenantManagerAuthUrl().isEmpty() ||
+                    properties.getTenantManagerClientId().isEmpty() ||
+                    properties.getTenantManagerClientSecret().isEmpty()) {
+                throw new DeploymentException("Unsupported configuration, \"registry.enable.multitenancy\" is enabled " +
+                        "\"registry.enable.auth\" is enabled " +
+                        "but the no auth properties aren't properly configured");
+            }
+
+            String tenantManagerAuthServerUrl = properties.getTenantManagerAuthUrl().get();
+
+            if (resolveIdentityServer) {
+                resolverHttpClient = new JdkHttpClientProvider().create(resolverRequestBasePath, Collections.emptyMap(), null, new AuthErrorHandler());
+                final IdentityServerResolver.SsoProviders ssoProviders = resolverHttpClient.sendRequest(getSSOProviders(resolverRequestPath));
+                if (!tenantManagerAuthServerUrl.equals(ssoProviders.getTokenUrl())) {
+                    tenantManagerAuthServerUrl = ssoProviders.getTokenUrl();
+                }
+            }
+
+            ApicurioHttpClient httpClient = new JdkHttpClientProvider().create(tenantManagerAuthServerUrl, Collections.emptyMap(), null, new AuthErrorHandler());
+
+            Duration tokenExpirationReduction = null;
+            if (properties.getTenantManagerAuthTokenExpirationReductionMs().isPresent()) {
+                log.info("Using configured tenant-manager auth token expiration reduction {}", properties.getTenantManagerAuthTokenExpirationReductionMs().get());
+                tokenExpirationReduction = Duration.ofMillis(properties.getTenantManagerAuthTokenExpirationReductionMs().get());
+            }
+
+            auth = new OidcAuth(httpClient,
+                    properties.getTenantManagerClientId().get(),
+                    properties.getTenantManagerClientSecret().get(),
+                    tokenExpirationReduction);
+        }
+
+        return new TenantManagerClientImpl(properties.getTenantManagerUrl().get(), clientConfigs, auth);
+    }
+
+    private TenantManagerClient getClient() {
```

Comment:
This can return `Optional` to make the case when multi-tenancy is not enabled explicit.